hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/ReadableFreezeUpgradeActions.java

```diff
@@ -179,6 +201,84 @@ private void extractAndReplaceArtifacts(
+
+        try (FileWriter fw = new FileWriter(configTxt.toFile());
+                BufferedWriter bw = new BufferedWriter(fw)) {
+            Streams.stream(nodeIds)
+                    .map(nodeId -> nodeStore.get(nodeId.number()))
+                    .filter(node -> node != null && !node.deleted())
+                    .sorted(Comparator.comparing(Node::nodeId))
+                    .forEach(node -> writeConfigLineAndPem(node, bw, artifactsLoc));
+        } catch (final IOException e) {
+            log.error("Failed to generate {} with exception : {}", configTxt, e);
+        }
+    }
+
+    private void writeConfigLineAndPem(
+            @NonNull final Node node, @NonNull final BufferedWriter bw, @NonNull final Path pathToWrite) {
+        requireNonNull(node);
+        requireNonNull(bw);
+        requireNonNull(pathToWrite);
+
+        var line = new StringBuilder();
+        int weight = 0;
+        final var name = "node" + node.nodeId();
+        final var alias = nameToAlias(name);
+        final var pemFile = pathToWrite.resolve("s-public-" + alias + ".pem");
+        final int INT = 0;
+        final int EXT = 1;
+
+        final var stakingNodeInfo = stakingInfoStore.get(node.nodeId());
+        if (stakingNodeInfo != null) {
+            weight = stakingNodeInfo.weight();
+        }
+
+        final var gossipEndpoints = node.gossipEndpoint();
+        if (gossipEndpoints.size() > 1) {
+            line.append("address, ")
+                    .append(node.nodeId())
+                    .append(", ")
+                    .append(node.nodeId()) // nodeId as nickname
+                    .append(", ")
+                    .append(name)
+                    .append(", ")
+                    .append(weight)
+                    .append(", ")
+                    .append(gossipEndpoints.get(INT).ipAddressV4().asUtf8String())
+                    .append(", ")
+                    .append(gossipEndpoints.get(INT).port())
+                    .append(", ")
+                    .append(gossipEndpoints.get(EXT).ipAddressV4().asUtf8String())
+                    .append(", ")
+                    .append(gossipEndpoints.get(EXT).port())
+                    .append(", ")
+                    .append(node.accountId().shardNum() + "." + node.accountId().realmNum() + "."
+                            + node.accountId().accountNum())
+                    .append("\n");
+            try {
+                bw.write(line.toString());
+            } catch (IOException e) {
+                log.error("Failed to write line {}} with exception : {}", line, e);
+            }
+            try {
+                Files.write(pemFile, node.gossipCaCertificate().toByteArray());
+            } catch (IOException e) {
+                log.error("Failed to write to {}} with exception : {}", pemFile, e);
+            }
+        } else {
+            log.error("Node has {}} gossip endpoints, expected greater than 1", gossipEndpoints.size());
+        }
```

Comment:
Is this an error? Then why do we have a `repeated` field while creating the node in protobufs?

Reply:
Typo. Corrected.